### PR TITLE
Add CritiqueBrainz as external OAuth Service

### PR DESCRIPTION
**Pull request by amCap1712**
Add option for users to link their CritiqueBrainz so that they can write mini-reviews.

### CODE REVIEW

File: listenbrainz/webserver/templates/user/music_services.html [hunk 1]
```diff
@@ -65,6 +65,40 @@ <h3>A note about permissions</h3>
             </div>
         </div>
 
+        <div class="panel panel-default">
```
Comment (amCap1712) [inline listenbrainz/webserver/templates/user/music_services.html:68]: Suggestions for what to show to users when asking them to link their CB account? I have added a bare bones line the benefit is they can write mini revies. But this should explain it a bit more.

**Comment by amCap1712** [comment]
Tested on bono and works as expected.

File: listenbrainz/webserver/templates/user/music_services.html [hunk 2]
```diff
@@ -65,6 +65,40 @@ <h3>A note about permissions</h3>
...
+                <div class="music-service-selection">
+                    <form action="{{ url_for('profile.music_services_disconnect', service_name='critiquebrainz') }}"
+                          method="post">
+                        {{
+                                service_permission_button(
+                                "critiquebrainz",
+                                current_critiquebrainz_permissions,
+                                "review",
+                                "Reviews",
+                                "You will be able to write mini-reviews of listena from ListenBrainz.")
```
Comment (jdaok) [inline listenbrainz/webserver/templates/user/music_services.html:87]:
```suggestion
                                "You will be able to publish mini-reviews for your Listens directly from ListenBrainz.")
```

File: listenbrainz/webserver/templates/user/music_services.html [hunk 3]
```diff
@@ -65,6 +65,40 @@ <h3>A note about permissions</h3>
...
+                        }}
+                        {{
+                                service_permission_button(
+                                "critiquebrainz",
+                                current_critiquebrainz_permissions,
+                                "disable",
+                                "Disable",
+                                "You will not be able to write mini-reviews of listena from ListenBrainz.")
```
Comment (jdaok) [inline listenbrainz/webserver/templates/user/music_services.html:95]:
```suggestion
                                "You will not be able to publish mini-reviews for your Listens directly from ListenBrainz.")
```

File: listenbrainz/webserver/templates/user/music_services.html (hunk 1 above)
Comment (jdaok) [reply listenbrainz/webserver/templates/user/music_services.html:68]: Adding onto alastairp's suggestion, we could set it to:

Connect to your CritiqueBrainz account to publish reviews for your **Listens** directly from ListenBrainz. 
Your reviews will be independently visible on CritiqueBrainz and appear publicly on your **CritiqueBrainz profile** unless removed. To view or delete your reviews, visit your **CritiqueBrainz profile.**

Comment (mayhem) [reply listenbrainz/webserver/templates/user/music_services.html:68]: :+1:

**Review by mayhem — APPROVED** [review]
Sorry for leaving this one hanging, I had done most of the review, but then lost connectivity. Looks good, thanks to @jdaok for helping with the english text!

File: listenbrainz/webserver/templates/user/music_services.html (hunk 1 above)
Comment (amCap1712) [reply listenbrainz/webserver/templates/user/music_services.html:68]: Fixed.

**Pull request merged**